client/src/App.jsx

```diff
@@ -82,29 +84,30 @@ class App extends React.Component {
 
     return (
       <Router>
-        <Switch>
- 
-          <Route exact path="/login">
-            <Login login={this.handleLogin} redirect={this.handleRedirect}/>
-          </Route>
-
-          <Route exact path="/register">
-            <Register login={this.handleLogin}/>
-          </Route>
-
-          <PrivateRoute exact path="/" component={Main} userId={id} />
-          <PrivateRoute exact path="/driver" component={Driver} userId={id} />
-          <PrivateRoute exact path="/rider" component={Rider} userId={id} />
-          <PrivateRoute exact path="/balance-update" component={BalanceUpdate} userId={id} />
-          <PrivateRoute exact path="/balance-transfer" component={BalanceTransfer} userId={id} />
-
-          <Route path="/">
-            <div>404 Not Found</div>
-          </Route>
-
-        </Switch>
-      </Router>
-
+          <Switch>
+  
+            <Route exact path="/login">
+              <Login login={this.handleLogin} redirect={this.handleRedirect}/>
+            </Route>
+
+            <Route exact path="/register">
+              <Register login={this.handleLogin}/>
+            </Route>
+
+          <AuthContext.Provider value={this.state} >
```

Comment:
I've always heard of useContext and never really understood what it does. Now I do. Thank you!